File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/entitytype/search/SampleTypeSearchCriteria.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.entitytype.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.entitytype.search.SampleTypeSearchCriteria")
public class SampleTypeSearchCriteria extends EntityTypeSearchCriteria
{

    private static final long serialVersionUID = 1L;

    public ListableSampleTypeSearchCriteria withListable() {
        return with(new ListableSampleTypeSearchCriteria());
    }
    
    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("SAMPLES_TYPE");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/entitytype/search/SampleTypeSearchCriteria.js
/**
 * @author pkupczyk
 */
define([ "require", "stjs", "as/dto/entitytype/search/EntityTypeSearchCriteria", "as/dto/entitytype/search/ListableSampleTypeSearchCriteria" ], 
	function(require, stjs, EntityTypeSearchCriteria) {
		var SampleTypeSearchCriteria = function() {
			EntityTypeSearchCriteria.call(this);
		};
		stjs.extend(SampleTypeSearchCriteria, EntityTypeSearchCriteria, [ EntityTypeSearchCriteria ], function(constructor, prototype) {
			prototype['@type'] = 'as.dto.entitytype.search.SampleTypeSearchCriteria';
			constructor.serialVersionUID = 1;
			prototype.withListable = function() {
				var ListableSampleTypeSearchCriteria = require("as/dto/entitytype/search/ListableSampleTypeSearchCriteria");
				return this.addCriteria(new ListableSampleTypeSearchCriteria());
			};
		}, {});
		
	return SampleTypeSearchCriteria;
})